### DISCUSSION TITLE
Can I use plugins when using the Assistants API？

### DISCUSSION THREAD

**Discussion by ymuichiro** [microsoft/semantic-kernel#9917] [Q&A] [answered]
ChatCompletionAgent can recognize plugins by using PromptExecutionSettings as before.
However, when using AzureAssistantAgent, etc., I do not know how to make it recognize plugins.

However, is there any sample code to use a plugin with AzureAssistantAgent?

**Comment by moonbox3** [comment, marked as answer]
Hi @ymuichiro, here's a sample showing how to use an AzureAssistantAgent with a plugin: https://github.com/microsoft/semantic-kernel/blob/main/python/samples/getting_started_with_agents/step7_assistant.py.

**Reply by moonbox3** [reply]
Note, @ymuichiro, we have a work item to allow one to pass in execution settings to the assistant agent -- right now, it'll use whatever is registered on the kernel.

**Reply by ymuichiro** [reply]
Hi @moonbox3 

Thank you! The following code worked. It seems it is used automatically internally.

```
kernel = Kernel()
kernel.add_plugin(TimePlugin(), "time")

agent = await AzureAssistantAgent.create(
    id="asst_xxxxxxxxxxxxxxxxxx",
    service_id=app_settings.SERVICE_ID,
    kernel=kernel,
    api_key=app_settings.OPENAI_COMPLETION_API_KEY,
    deployment_name=app_settings.OPENAI_COMPLETION_DEPLOYMENT_NAME,
    api_version="2024-05-01-preview",
    endpoint=Url(app_settings.OPENAI_COMPLETION_ENDPOINT),
)

thread = await agent.create_thread()
await agent.add_chat_message(thread, ChatMessageContent(role=AuthorRole.USER, content="what time is it now?"))

async for message_content in agent.invoke(thread):
    print(message_content.content)
```

**Reply by moonbox3** [reply]
Great, thanks for your reply and confirmation. As mentioned, please keep in mind that the `FunctionChoiceBehavior` construct isn't currently available (yet). We'll need to add that functionality in the coming weeks, along with being able to pass in KernelArguments and such.
